soc/ti/k3/am6x/common/ctrl_partitions.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2025 Texas Instruments Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/util.h>
+#include <zephyr/arch/common/sys_io.h>
+#include <zephyr/drivers/mm/system_mm.h>
+
+#define KICK0_OFFSET     (0x1008)
+#define KICK1_OFFSET     (0x100C)
+#define KICK0_UNLOCK_VAL (0x68EF3490U)
+#define KICK1_UNLOCK_VAL (0xD172BC5AU)
+#define KICK_LOCK_VAL    (0x0U)
+
+const uintptr_t ctrl_partitions[] = {
+#if defined CONFIG_SOC_AM6234_M4 || defined CONFIG_SOC_AM6442_M4
+	0x4080000, /* mcu/wkup padcfg 1 */
+	0x4084000, /* mcu/wkup padcfg 2 */
```

Comment:
Instead of keeping raw addresses here, can you add a define and also derive the second page address from the first using a macro? ```base + partition * CTRL_MMR0_PARTITION_SIZE```